src/main/scala/mesosphere/marathon/api/HttpModule.scala

```diff
@@ -0,0 +1,193 @@
+      case (Some(_), Some(_)) =>
+        logger.warn("Both HTTP and HTTPS support have been configured. " +
+          s"Consider disabling HTTP with --${conf.disableHttp.name}")
+      case (None, None) =>
+        throw new IllegalArgumentException(
+          "Invalid configuration: Neither HTTP nor HTTPS support has been configured.")
+      case _ =>
+      // everything seems fine
+    }
+
+    if (conf.httpCompression()) {
+      val gzipHandler = new GzipHandler()
+      gzipHandler.addExcludedMimeTypes("text/event-stream") //exclude event stream compression
+      gzipHandler.setHandler(handlerCollection)
+      server.setHandler(gzipHandler)
+    } else {
+      server.setHandler(handlerCollection)
+    }
+
+    server
+  }
+
+  private def getHTTPConnector(server: Server, httpConfig: HttpConfiguration): Option[ServerConnector] = {
+    if (!conf.disableHttp()) {
+      val connector = new ServerConnector(server, new HttpConnectionFactory(httpConfig))
+      configureConnectorAddress(connector, conf.httpAddress, conf.httpPort)
+      Some(connector)
+    } else {
+      None
+    }
+  }
+
+  private def getHTTPSConnector(server: Server, httpConfig: HttpConfiguration): Option[ServerConnector] = {
+    def createHTTPSConnector(keystorePath: String, keystorePassword: String): ServerConnector = {
+      val keystore = new File(keystorePath)
+      require(
+        keystore.exists() && keystore.canRead,
+        f"${conf.sslKeystorePath()} is invalid or not readable!")
+
+      val contextFactory = new SslContextFactory()
+      contextFactory.setKeyStorePath(keystorePath)
+      contextFactory.setKeyStorePassword(keystorePassword)
+
+      val sslConfig = new HttpConfiguration(httpConfig)
+      sslConfig.addCustomizer(new SecureRequestCustomizer())
+
+      val sslConnector = new ServerConnector(server, new SslConnectionFactory(contextFactory, HttpVersion.HTTP_1_1.asString()), new HttpConnectionFactory(sslConfig))
+      configureConnectorAddress(sslConnector, conf.httpsAddress, conf.httpsPort)
+
+      sslConnector
+    }
+
+    for {
+      keystorePath <- conf.sslKeystorePath.get
+      keystorePassword <- conf.sslKeystorePassword.get
+      connector = createHTTPSConnector(keystorePath, keystorePassword)
+    } yield connector
+  }
+
+  private[this] def configureConnectorAddress(connector: ServerConnector, addressOpt: ScallopOption[String], portOpt: ScallopOption[Int]): Unit = {
+    connector.setIdleTimeout(30000)
+    addressOpt.foreach(connector.setHost)
+    portOpt.get match {
+      case Some(port) =>
+        connector.setPort(port)
+      case None =>
+        // shouldn't happen because our port configurations all have defaults
+        throw new IllegalArgumentException("Port required.")
+    }
+  }
+
+  lazy val requestLogHandler: RequestLogHandler = {
+    val handler = new RequestLogHandler()
+    handler.setRequestLog(requestLog)
+    handler
+  }
+
+  lazy val handler: ServletContextHandler = {
```

Comment:
Could you add some doc string. What handler is this?

Reply:
lol came from Chaos. I'll do my best.